mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp

```diff
@@ -1922,7 +1923,14 @@ transform::PadOp::apply(transform::TransformRewriter &rewriter,
         continue;
       }
       // Otherwise, add the attribute directly.
-      if (attr.getType() != elementType) {
+      if (isa<TypedAttr>(attr) &&
+          cast<TypedAttr>(attr).getType() != elementType) {
```

Comment:
`isa` followed by `cast` is an explicit anti-pattern in LLVM, please use `dyn_cast` combined with c++17 feature of declaring a value inside `if`.